column.go

```diff
@@ -488,7 +488,7 @@ func schemaRepetitionTypeOf(s *format.SchemaElement) format.FieldRepetitionType
 }
 
 func (c *Column) decompress(compressedPageData []byte, uncompressedPageSize int32) (page *buffer, err error) {
-	page = uncompressedPageBufferPool.get()
+	page = uncompressedPageBufferPool.get(int(uncompressedPageSize))
 	if uncompressedPageSize > 0 {
 		page.resize(int(uncompressedPageSize))
 	}
```

Comment:
I'm curious what you'd think about potentially removing the `resize` method? We are now passing the size to `get` which could set the buffer size accordingly already?

Reply:
my assumption was that the code used `resize()` method aside from immediately after pulling from the pool, but that doesn't seem to be the case. a quick attempt to remove this caused a failing test. i'll give it a real try early next week.

Reply:
oh, one other question. if `get()` is supposed to start doing the work of resize do we see value in keeping the code to align slices to 4096 bytes?

Reply:
🤔 aligning the allocations to the size class seems like the right approach to me, do you have any concerns?